python/ray/data/impl/table_block.py

```diff
@@ -0,0 +1,231 @@
+import collections
+
+from typing import Dict, Iterator, List, Union, Tuple, Any, TypeVar
+
+from ray.data.block import Block, BlockAccessor
+from ray.data.impl.block_builder import BlockBuilder
+from ray.data.impl.simple_block import SimpleBlockBuilder
+from ray.data.impl.size_estimator import SizeEstimator
+
+T = TypeVar("T")
+
+# An Arrow block can be sorted by a list of (column, asc/desc) pairs,
+# e.g. [("column1", "ascending"), ("column2", "descending")]
+SortKeyT = List[Tuple[str, str]]
+GroupKeyT = Union[None, str]
+
+# The max size of Python tuples to buffer before compacting them into a
+# table in the BlockBuilder.
+MAX_UNCOMPACTED_SIZE_BYTES = 50 * 1024 * 1024
+
+
+class TableRow:
```

Comment:
Use ABCs and abstractmethod.

Reply:
There are existing classes that don't use ABC, e.g. `BlockAccessor`, and I don't know why.